src/web/WebSchedulerService.cpp

```diff
@@ -84,75 +83,107 @@ StateUpdateResult WebScheduler::update(JsonObject & root, WebScheduler & webSche
+    JsonObject                                 output = doc_output.to<JsonObject>();
+
+    // prefix "api/" to command string
+    char command_str[100];
+    snprintf(command_str, sizeof(command_str), "/api/%s", cmd);
+    uint8_t return_code = Command::process(command_str, true, input, output); // admin set
+
+    if (return_code == CommandRet::OK) {
+        EMSESP::logger().debug("Scheduled command %s with data %s successfully", cmd, data);
+        if (strlen(data) == 0 && Mqtt::enabled() && Mqtt::send_response() && output.size()) {
+            Mqtt::queue_publish("response", output);
+        }
+        return true;
+    }
+
+    char error[100];
+    if (output.size()) {
+        snprintf(error,
+                 sizeof(error),
+                 "Scheduled command %s failed with error: %s (%s)",
+                 cmd,
+                 (const char *)output["message"],
+                 Command::return_code_string(return_code).c_str());
+    } else {
+        snprintf(error, sizeof(error), "Scheduled command %s failed with error code (%s)", cmd, Command::return_code_string(return_code).c_str());
+    }
+    emsesp::EMSESP::logger().err(error);
+    return false;
+}
+
 // process any scheduled jobs
 // checks on the minute
 void WebSchedulerService::loop() {
+    // initialize static value on startup
+    static int8_t   last_tm_min     = -1; // invalid value also used for startup commands
+    static uint32_t last_uptime_min = 0;
+
     // get list of scheduler events and exit if it's empty
     EMSESP::webSchedulerService.read([&](WebScheduler & webScheduler) { scheduleItems = &webScheduler.scheduleItems; });
     if (scheduleItems->size() == 0) {
         return;
     }
 
-    time_t          now            = time(nullptr);
-    tm *            tm             = localtime(&now);
-    static uint32_t last_min       = 100;                    // some high marker
-    uint32_t        uptime_seconds = uuid::get_uptime_sec(); // sync to EMS-ESP clock
-    uint32_t        uptime_min     = uptime_seconds / 60;    // sync to EMS-ESP clock
-
-    // check if we're on the minute - taking the EMS-ESP uptime
-    // with the exception of allowing the first pass (on boot) through
-    if (uptime_min == last_min || tm->tm_year < 122) {
-        return;
+    // check startup commands
+    if (last_tm_min == -1) {
+        for (ScheduleItem & scheduleItem : *scheduleItems) {
+            if (scheduleItem.active && scheduleItem.flags == SCHEDULEFLAG_SCHEDULE_TIMER && scheduleItem.elapsed_min == 0) {
+                scheduleItem.retry_cnt = command(scheduleItem.cmd.c_str(), scheduleItem.value.c_str()) ? 0xFF : 0;
+            }
+        }
+        last_tm_min = 0; // startup done, now use for RTC
+    }
+    // check timer every minute, sync to EMS-ESP clock
+    uint32_t uptime_min = uuid::get_uptime_sec() / 60;
+    if (last_uptime_min != uptime_min || last_uptime_min == -1) {
```

Comment:
last_uptime_min is unsigned so this will always be false

Reply:
Ups, yes, can be removed, left from a first try to put startup and timer in one for().